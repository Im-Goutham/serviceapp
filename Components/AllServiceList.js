import React, {Component} from 'react';
import {StyleSheet, ListView, Text, View,Image,TouchableOpacity} from 'react-native';
import { connect } from 'react-redux';
import * as actions from '../actions';
import LinearGradient from 'react-native-linear-gradient';

class AllServiceList extends Component {
    constructor() {
        super();
        var ds = new ListView.DataSource({rowHasChanged: (row1, row2) => true});
        this.state = {
            dataSource: ds.cloneWithRows([
                {text: 'list 1'},
                {text: 'list 2'},
                {text: 'list 3'},
                {text: 'list 4'},
                {text: 'list 5'}
                ]),
            sectionID: null,
            rowID: null,
            images:[
                {image: require('../images/service1.png')},
                {image: require('../images/service2.png')},
                {image: require('../images/service3.png')},
            ],
            imgIndex: 0,
            photoShow:false
        };
    }


    _renderRow(rowData, sectionID, rowID) {

        return (
                <View style={styles.servicesBox} >
                    <View style={{backgroundColor:"transparent", paddingTop: 15,  }}>
                        <View style={{flex:1,flexDirection:'row', paddingHorizontal:15}}>
                            <View style={{flex:2,justifyContent:'flex-start',alignItems:'flex-start'}}>
                                <View style={styles.imageShadow}>
                                    <Image source={require('../images/svp1.png')} style={[styles.img_placeholder,{borderRadius:35,width:70,height:70}]}/>
                                    <Image source={require('../images/check.png')} style={styles.check}/>
                                </View>
                            </View>
                            <View style={{flex:7,paddingHorizontal:20,paddingBottom:10,justifyContent:'space-between'}}>
                                <View style={{flex:1}}>
                                    <Text style={{fontSize:17,color:'#4A4A4A'}}>Clayton L.</Text>
                                     <View style={{flexDirection:'row',paddingVertical:5}}>
                                       <Image source={require('../assets/icons/star_gold.png')} style={{width:15,height:15}} resizeMode="contain" resizeMethod="resize" />
                                       <Text style={{color:'#4A4A4A',fontFamily: 'Montserrat-Bold',paddingLeft:6}}>5</Text>
                                     </View>
                                     <View style={{flexDirection:'row',paddingVertical:5}}>
                                       <Image source={require('../assets/icons/calender.png')} style={{width:15,height:15}} resizeMode="contain" resizeMethod="resize" />
                                       <Text style={{color:'#4A4A4A',fontFamily: 'Montserrat-Regular',paddingLeft:10}}>Appt. On 08 Nov 2018</Text>
                                     </View>
                                    <Text style={{fontFamily:'Montserrat-Medium',fontSize:14,color:'#9B9B9B'}}>Plumber & 5 More</Text>
                                </View>
                            </View>
                        </View>
                        <View style={{flexDirection:"row", justifyContent:'space-between'}}>
                            <TouchableOpacity style={{ width : "30%", backgroundColor:"#CCCCCC", height : 40, borderBottomLeftRadius:10,justifyContent:"center", alignItems:"center"}}>
                                <LinearGradient
                                        colors={['#F2F2F2', '#CCCCCC']}
                                        start={{x: 0, y: 0}}
                                        end={{x: 1, y: 0}}
                                        style={{height : 40, backgroundColor:"transparent", width : "100%", borderBottomRightRadius:10,  justifyContent:"center", alignItems:"center"}}>
                                        <Text style={{color:"black", fontFamily:"Montserrat-Regular", fontSize:15}}>REJECT</Text>
                                    </LinearGradient>
                            </TouchableOpacity>
                            <TouchableOpacity style={{ width : "70%"}}>
                                <LinearGradient
                                    colors={['#3E85EF', '#3EBDEF']}
                                    start={{x: 0, y: 0}}
                                    end={{x: 1, y: 0}}
                                    style={{height : 40, backgroundColor:"transparent", width : "100%", borderBottomRightRadius:10,  justifyContent:"center", alignItems:"center"}}>
                                    <Text style={{color:"#fff", fontFamily:"Montserrat-Bold", fontSize:15}}>ACCEPT</Text>
                                </LinearGradient>
                            </TouchableOpacity>
                        </View>
                    </View>
                </View>
        );
    }
    showImage(index){
        console.log('imgIndex',index);
        let {images} = this.state;
        let imagesArray = images.map((image)=>{
            return {url:image}
        })
        this.props.showPhotoView({index,images:imagesArray,photoShow:true})
    }
    render() {
        let {images,imgIndex,photoShow} = this.state;
        return (
            <View style={{flex:1,backgroundColor:"rgb(249,252,255)"}} testID="servicesBox">
                <ListView
                    testID='indianic'
                    scrollEnabled
                    dataSource={this.state.dataSource}
                    renderRow={this._renderRow.bind(this)}
                    style={styles.listview}
                />
            </View>
        );
    }
}

var styles = StyleSheet.create({
    listview: {
        flex: 1,
    },
    liText: {
        color: '#333',
        fontSize: 16,
    },
    button:{
        width: '100%',
        borderRadius:30,
        paddingVertical:10
    },
    btnText: {
        textAlign:'center',
        color:'white',
        fontFamily: 'Montserrat-Bold',
        fontSize:15
    },
    imgBox: {
        flex:1,
        padding:1,
        shadowOffset: { width: 0, height: 2 },
        shadowOpacity: 0.2,
        shadowRadius: 2,
        elevation: 3,
    },
    overlay: {
        position: 'absolute',
        top: 0,
        right: 0,
        bottom: 0,
        left: 0,
        borderRadius:5,
        backgroundColor: 'rgba(0,0,0,.6)',
        opacity: 2,
        justifyContent:'center',
        alignItems:'center'
    },
    imgStyle:{
        width:100,
        height:100,
    },
    servicesBox: {
        flex: 1,
        marginVertical: 5,
        marginHorizontal: 10,
        // paddingVertical: 25,
        // paddingHorizontal:20,
        borderRadius:10,
        backgroundColor:'white',
        shadowOffset: { width: 0, height: 2 },
        shadowOpacity: 0.2,
        shadowRadius: 2,
        elevation: 3,
    },
    imageShadow: {
        shadowOffset: { width: 0, height: 3 },
        shadowOpacity: 0.3,
        shadowRadius: 2,
        elevation: 3,
    },
    img_placeholder: {
        width: 80,
        height: 80,
        borderRadius:5,
        position: 'relative',
        top: 0,
        left: 0
    },
    check: {
        width: 18,
        height: 18,
        borderRadius:9,
        position: 'absolute',
        bottom: 0,
        left: 0
    },
    iconStyle: {
        width:15,
        height:15
    }
});
export default connect(null, actions)(AllServiceList);
