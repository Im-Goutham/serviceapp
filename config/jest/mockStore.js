export const initialState = {
    user: {
      user: {},
      backButton: false
    },
    photo: {
      photoView: false,
      images: [],
      index: 0
    }
  }