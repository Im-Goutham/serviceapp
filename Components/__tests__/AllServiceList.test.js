jest.unmock('redux-mock-store')
jest.unmock('redux-thunk')

import React, {Component} from 'react';
import { View, Text, StyleSheet, ListView } from 'react-native';
import { shallow } from 'enzyme';
import configureMockStore from 'redux-mock-store';
import AllServiceList from '../AllServiceList';
import {initialState} from '../../config/jest/mockStore';

import thunk from 'redux-thunk';
let store;
const middlewares = [thunk];
const mockStore = configureMockStore(middlewares);

function sum(a, b) {
  return a + b;
}

describe('AllServiceList', () => {
    let wrapper;
    beforeEach(() => {
        store = mockStore(initialState);
        wrapper = shallow(<AllServiceList store={store}/>).dive();
    })
    it('it should render 1 view component', () => {
        expect(wrapper.find(ListView)).toBeDefined();
    });
    it('should return the sum of two numbers (4 + 5 = 9)', () => {
      expect(sum(4, 5)).toEqual(9);
    });
});
