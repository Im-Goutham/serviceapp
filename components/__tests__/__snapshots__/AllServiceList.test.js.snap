// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AllServiceList Snapshot testing should have same snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <AllServiceList
    hidePhotoView={[Function]}
    navigation={
      Object {
        "goBack": [MockFunction],
        "navigate": [MockFunction],
      }
    }
    onPress={[MockFunction]}
    showBackButton={[Function]}
    showPhotoView={[Function]}
    signIn={[Function]}
    store={
      Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <ListViewMock
        dataSource={
          ListViewDataSource {
            "items": 5,
          }
        }
        renderRow={[Function]}
        renderScrollComponent={[Function]}
        scrollEnabled={true}
        style={
          Object {
            "flex": 1,
          }
        }
        testID="indianic"
      />,
      "style": Object {
        "backgroundColor": "rgb(249,252,255)",
        "flex": 1,
      },
      "testID": "servicesBox",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "dataSource": ListViewDataSource {
          "items": 5,
        },
        "renderRow": [Function],
        "renderScrollComponent": [Function],
        "scrollEnabled": true,
        "style": Object {
          "flex": 1,
        },
        "testID": "indianic",
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <ListViewMock
          dataSource={
            ListViewDataSource {
              "items": 5,
            }
          }
          renderRow={[Function]}
          renderScrollComponent={[Function]}
          scrollEnabled={true}
          style={
            Object {
              "flex": 1,
            }
          }
          testID="indianic"
        />,
        "style": Object {
          "backgroundColor": "rgb(249,252,255)",
          "flex": 1,
        },
        "testID": "servicesBox",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "dataSource": ListViewDataSource {
            "items": 5,
          },
          "renderRow": [Function],
          "renderScrollComponent": [Function],
          "scrollEnabled": true,
          "style": Object {
            "flex": 1,
          },
          "testID": "indianic",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
